src/helpers/survey.js

```diff
@@ -0,0 +1,239 @@
+import axios from 'axios';
+import _merge from 'lodash/merge';
+import { BufferLoader, AudioContext } from '@/helpers/audio';
+import { LEVANTE_SURVEY_RESPONSES_KEY } from '@/constants/bucket';
+
+export const fetchAudioLinks = async (surveyType) => {
+  const response = await axios.get('https://storage.googleapis.com/storage/v1/b/road-dashboard/o/');
+  const files = response.data || { items: [] };
+  const audioLinkMap = {};
+  files.items.forEach((item) => {
+    if (item.contentType === 'audio/mpeg' && item.name.startsWith(surveyType)) {
+      const splitParts = item.name.split('/');
+      const fileLocale = splitParts[1];
+      const fileName = splitParts.at(-1).split('.')[0];
+      if (!audioLinkMap[fileLocale]) {
+        audioLinkMap[fileLocale] = {};
+      }
+      audioLinkMap[fileLocale][fileName] = `https://storage.googleapis.com/road-dashboard/${item.name}`;
+    }
+  });
+  return audioLinkMap;
+};
+
+export function getParsedLocale(locale) {
+  return (locale || '').split('-')?.[0] || 'en';
+}
+
+function finishedLoading({ bufferList, parsedLocale, setSurveyAudioLoading, setSurveyAudioPlayerBuffers }) {
+  setSurveyAudioPlayerBuffers(parsedLocale, bufferList);
+  setSurveyAudioLoading(false);
+}
+
+// Function to fetch buffer or return from the cache
+export const fetchBuffer = ({
+  parsedLocale,
+  setSurveyAudioLoading,
+  audioLinks,
+  surveyAudioBuffers,
+  setSurveyAudioPlayerBuffers,
+}) => {
+  const context = new AudioContext();
+  // buffer already exists for the given local
+  if (surveyAudioBuffers[parsedLocale]) {
+    return;
+  }
+  setSurveyAudioLoading(true);
+  const bufferLoader = new BufferLoader(context, audioLinks[parsedLocale], (bufferList) =>
+    finishedLoading({ bufferList, parsedLocale, setSurveyAudioLoading, setSurveyAudioPlayerBuffers }),
+  );
+
+  bufferLoader.load();
+};
+
+export const showAndPlaceAudioButton = ({ playAudioButton, el }) => {
+  if (playAudioButton) {
+    playAudioButton.classList.add('play-button-visible');
+    playAudioButton.style.display = 'flex';
+    el.appendChild(playAudioButton);
+  }
+};
+
+export function restoreSurveyData({ surveyInstance, uid, selectedAdmin, surveyResponsesData, surveyStore }) {
+  // Try to get data from localStorage first
+  const prevData = window.localStorage.getItem(`${LEVANTE_SURVEY_RESPONSES_KEY}-${uid}`);
```

Comment:
As you're accessing the localStorage in several places in this file, best to move this to a separate method: 

```js
const getSurveyResponsesData = (userId) => {
  const surveyResponsesData = window.localStorage.getItem(`${SURVEY_RESPONSES_QUERY_KEY}-${userId}`)
  return surveyResponsesData ? JSON.parse(surveyResponsesData) : null
}

const setSurveyResponseData = (userId, data) => {
  window.localStorage.setItem(`${LEVANTE_SURVEY_RESPONSES_KEY}-${userId}`, JSON.stringify(data));
}